Error while checking build/O2Physics/o2 for 7c52a01459f256856d50d946a3e6ea28e8b4a52f at 2023-05-10 07:44:
```
## sw/BUILD/O2Physics-latest/log
/sw/SOURCES/O2Physics/2563-slc7_x86-64/0/PWGMM/Mult/Tasks/dndeta.cxx:529:42: error: passing 'const o2::soa::Filtered<o2::soa::Join<o2::soa::Join<o2::aod::McParticles_001Extension, o2::soa::Table<o2::soa::Index<0, -1>, o2::aod::mcparticle::McCollisionId, o2::aod::mcparticle::PdgCode, o2::aod::mcparticle::StatusCode, o2::aod::mcparticle::Flags, o2::aod::mcparticle::MothersIds, o2::aod::mcparticle::DaughtersIdSlice, o2::aod::mcparticle::Weight, o2::aod::mcparticle::Px, o2::aod::mcparticle::Py, o2::aod::mcparticle::Pz, o2::aod::mcparticle::E, o2::aod::mcparticle::Vx, o2::aod::mcparticle::Vy, o2::aod::mcparticle::Vz, o2::aod::mcparticle::Vt, o2::aod::mcparticle::ProducedByGenerator<o2::aod::mcparticle::Flags>, o2::aod::mcparticle::FromBackgroundEvent<o2::aod::mcparticle::Flags>, o2::aod::mcparticle::GetGenStatusCode<o2::aod::mcparticle::Flags, o2::aod::mcparticle::StatusCode>, o2::aod::mcparticle::GetHepMCStatusCode<o2::aod::mcparticle::Flags, o2::aod::mcparticle::StatusCode>, o2::aod::mcparticle::GetProcess<o2::aod::mcparticle::Flags, o2::aod::mcparticle::StatusCode>, o2::aod::mcparticle::IsPhysicalPrimary<o2::aod::mcparticle::Flags> > >, o2::soa::Table<o2::aod::idx::TrackIds> > >' as 'this' argument discards qualifiers [-fpermissive]
/sw/slc7_x86-64/O2/nightly-20230503-local6/include/Framework/ASoA.h:2707:9: error: no match for 'operator*=' (operand types are 'o2::soa::FilteredBase<o2::soa::Join<o2::soa::Table<o2::aod::track::BestCollisionId, o2::aod::pwgmm::indices::TrackId, o2::aod::track::BestDCAXY, o2::aod::track::BestDCAZ>, o2::soa::Table<o2::aod::track::X, o2::aod::track::Alpha, o2::aod::track::Y, o2::aod::track::Z, o2::aod::track::Snp, o2::aod::track::Tgl, o2::aod::track::Signed1Pt, o2::aod::track::PtStatic, o2::aod::track::PStatic, o2::aod::track::EtaStatic, o2::aod::track::PhiStatic> > >' and 'const o2::soa::FilteredBase<o2::soa::Join<o2::soa::Table<o2::aod::track::BestCollisionId, o2::aod::pwgmm::indices::TrackId, o2::aod::track::BestDCAXY, o2::aod::track::BestDCAZ>, o2::soa::Table<o2::aod::track::X, o2::aod::track::Alpha, o2::aod::track::Y, o2::aod::track::Z, o2::aod::track::Snp, o2::aod::track::Tgl, o2::aod::track::Signed1Pt, o2::aod::track::PtStatic, o2::aod::track::PStatic, o2::aod::track::EtaStatic, o2::aod::track::PhiStatic> > >')
/sw/slc7_x86-64/O2/nightly-20230503-local6/include/Framework/ASoA.h:1119:35: error: could not convert '((const o2::framework::PresliceBase<o2::soa::Join<o2::soa::Table<o2::aod::track::BestCollisionId, o2::aod::pwgmm::indices::TrackId, o2::aod::track::BestDCAXY, o2::aod::track::BestDCAZ>, o2::soa::Table<o2::aod::track::X, o2::aod::track::Alpha, o2::aod::track::Y, o2::aod::track::Z, o2::aod::track::Snp, o2::aod::track::Tgl, o2::aod::track::Signed1Pt, o2::aod::track::PtStatic, o2::aod::track::PStatic, o2::aod::track::EtaStatic, o2::aod::track::PhiStatic> >, false>*)this)->o2::framework::PresliceBase<o2::soa::Join<o2::soa::Table<o2::aod::track::BestCollisionId, o2::aod::pwgmm::indices::TrackId, o2::aod::track::BestDCAXY, o2::aod::track::BestDCAZ>, o2::soa::Table<o2::aod::track::X, o2::aod::track::Alpha, o2::aod::track::Y, o2::aod::track::Z, o2::aod::track::Snp, o2::aod::track::Tgl, o2::aod::track::Signed1Pt, o2::aod::track::PtStatic, o2::aod::track::PStatic, o2::aod::track::EtaStatic, o2::aod::track::PhiStatic> >, false>::sliceInfo.o2::framework::SliceInfoUnsortedPtr::getSliceFor(value)' from 'span<const long int>' to 'span<long int>'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/O2Physics/2563/7c52a01459f256856d50d946a3e6ea28e8b4a52f/build_O2Physics_o2/pretty.html).

--- a/PWGMM/Mult/TableProducer/trackPropagation.cxx
+++ b/PWGMM/Mult/TableProducer/trackPropagation.cxx
@@ -35,20 +35,8 @@
 using SMatrix55 = ROOT::Math::SMatrix<double, 5, 5, ROOT::Math::MatRepSym<double, 5>>;
 using SMatrix5 = ROOT::Math::SVector<Double_t, 5>;
 
-// The Run 3 AO2D stores the tracks at the point of innermost update. For a
-// track with ITS this is the innermost (or second innermost) ITS layer. For a
-// track without ITS, this is the TPC inner wall or for loopers in the TPC even
-// a radius beyond that. In order to use the track parameters, the tracks have
-// to be propagated to the collision vertex which is done by this task. The task
-// consumes the TracksIU and TracksCovIU tables and produces Tracks and
-// TracksCov to which then the user analysis can subscribe.
-//
-// This task is not needed for Run 2 converted data.
-// There are two versions of the task (see process flags), one producing also
-// the covariance matrix and the other only the tracks table.
-
-// This is an alternative version of the propagation task with special treatment
-// of ambiguous tracks
+// This is a special version of the propagation task chosing the closest vertex
+// among the compatible, which is defined by track-to-collision-associator
 
 using namespace o2;
 using namespace o2::framework;
@@ -240,6 +228,5 @@ struct AmbiguousTrackPropagation {
 //****************************************************************************************
 WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
 {
-  WorkflowSpec workflow{adaptAnalysisTask<AmbiguousTrackPropagation>(cfgc)};
-  return workflow;
+  return {adaptAnalysisTask<AmbiguousTrackPropagation>(cfgc)};
 }

